feat(artifacts): update gitrepo configuration (#1941)
spinnaker/spinnaker.github.io: setup/artifacts/gitrepo.md

---
layout: single
title:  "Configure a Git Repo artifact account"
sidebar:
  nav: setup
---

{% include toc %}

> This shows how to configure a Git Repo artifact account so Spinnaker can use entire repositories as a single artifact.

Reviewer: ```suggestion
This shows how to configure a Git Repo artifact account so Spinnaker can use an entire repository as a single artifact.
```

Reviewer: Unless it's possible to (and plausible that one would want to) have an artifact comprising multiple repos.

Author: Nope, you're right, it would be just one 👍

## Prerequisites

* You need a Git account.

## Enable Git Repo artifacts

First, enable [artifact support](/reference/artifacts-with-artifactsrewrite//#enabling-artifact-support).

Next, enable the Git Repo artifact provider:

```bash
hal config artifact gitrepo enable
```

## Configure Auth
Choose to set up either token, user-password or ssh key auth below.

### Token Auth

> Start by generating an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.

Reviewer: ```suggestion
1. Generate an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.
```

> Place the token in a file (`$TOKEN_FILE`) readable by Halyard:

Reviewer: ```suggestion
1. Place the token in a file (`$TOKEN_FILE`) readable by Halyard:
```

> ```bash

Reviewer: ```suggestion
   ```bash
```

> echo $TOKEN > $TOKEN_FILE

Reviewer: ```suggestion
   echo $TOKEN > $TOKEN_FILE
```

> ```

Reviewer: ```suggestion
   ```
```

> Add an artifact account:

Reviewer: ```suggestion
1. Add an artifact account:
```

> ```bash

Reviewer: ```suggestion
   ```bash
```

> hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \

Reviewer: ```suggestion
   hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \
```

>     --token-file $TOKEN_FILE

Reviewer: ```suggestion
       --token-file $TOKEN_FILE
```

> ```

Reviewer: ```suggestion
   ```
```

### User-Password Auth

> The username-password file contents should be in the format:

Reviewer: ```suggestion
1. Create a username-password file, with contents in the following format:
```

Reviewer: (Similarly, number the steps and indent the command so the numbering works.)

 ```
 <username>:<password>
 ```

Add an artifact account:

```bash
hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \
    ----username-password-file $PASSWORD_FILE
```

### SSH Key Auth

> Add an artifact account:

Reviewer: And no need to number this one, because it's just the one directive.
